numpy/typing/tests/data/reveal/flatiter.py

```diff
@@ -14,4 +14,4 @@
 reveal_type(a[...])  # E: numpy.ndarray[Any, numpy.dtype[numpy.str_]]
 reveal_type(a[:])  # E: numpy.ndarray[Any, numpy.dtype[numpy.str_]]
 reveal_type(a.__array__())  # E: numpy.ndarray[Any, numpy.dtype[numpy.str_]]
-reveal_type(a.__array__(np.float64))  # E: numpy.ndarray[Any, numpy.dtype[Any]]
+reveal_type(a.__array__(np.dtype(np.float64)))  # E: numpy.ndarray[Any, numpy.dtype[{float64}]]
```

Comment:
One follow-up question: `numpy.dtype[{float64}]` is a little verbose. How hard would it be to make `numpy.ndarray[Any, numpy.float64]` work?